### ISSUE TITLE
Add syntax to define and use variables

### ISSUE THREAD

**Issue by DiddiLeija** [DiddiLeija/diddiparser2#18]
**Is your feature request related to a problem? Please describe.**

The DiddiScript functions are so great: they are extensible and dynamic. However, they depend on rough strings as parameters. SO they are a bit... restrictive.

**Describe the solution you'd like**

A modification to the grammar, so we can define variables like this:

```
my_var = "stuff";
```

And then, we can index them into string arguments. I've been thinking in something like:

```
some_function("I got a lot of !{my_var} to do");  !# indexing via "!{variable}"
```

**Describe alternatives you've considered**

Something like a `store_variable` function, and keep the indexing. However, doing that could be a bit glitchy.

**Additional context**

N/A

**Comment by DiddiLeija** [comment]
If #20 gets merged, I'll move this to a DSGP.

**Comment by DiddiLeija** [comment]
Closed in favor of #21 (since #20 enabled DSGPs, I created one to propose variables).
